storage/aptosdb/src/pruner/ledger_pruner_manager.rs

```diff
@@ -49,8 +52,8 @@ impl PrunerManager for LedgerPrunerManager {
         self.prune_window
     }
 
-    fn get_min_readable_version(&self) -> Option<Version> {
-        self.pruner_min_readable_version.lock().map(|x| x)
+    fn get_min_readable_version(&self) -> Version {
+        self.pruner.as_ref().lock().min_readable_version()
```

Comment:
This lock is locked for doing the actual pruning, which will make all API calls slow.. We need to share state between threads in finer granularity.. (maybe use a AtomicVersion?)

Reply:
We probably don't really need the mutex of pruner then as the min_readable_version and other versions are already atomic.

Reply:
Get rid of the mutex.

Reply:
Actually getting rid of the mutex makes some tests flaky https://github.com/aptos-labs/aptos-core/runs/7662989102?check_suite_focus=true. Will take a look.

Reply:
The reason of race condition is because we change min_readable_version before deleting the items from DBs and when we don't have mutex, the test thread could read the new min_readable_version and proceed with the verifying logics of items being deleted from the DB. To address the issue, added a hack to sleep after the min_readable_version >= target version and then do the DB check. Ideally we should use a channel and notify the testing thread when DB finish deleting the items but that would make the code pretty verbose just for the testing purpose.